aii-core/src/main/perl/Shellfe.pm

```diff
@@ -671,144 +679,112 @@ sub fetch_profiles
-        my $lock = $self->lock_node($node);
-        if (! $lock) {
-            $self->error("aii-shellfe: couldn't acquire lock on $node for install");
-            exit(1);
-        }
-        $self->run_plugin ($st, NBP, INSTALLMETHOD);
-    }
+    my ($self, $node, $st) = @_;
+    $self->run_plugin ($st, NBP, INSTALLMETHOD);
 }
 
 # Runs the Status method of the NBP plugins of the nodes given as
 # arguments.
-sub status
+sub _status
 {
-    my ($self, %nl) = @_;
-    while (my ($node, $st) = each (%nl)) {
-        $self->debug (1, "Showing the state of $node");
-        $self->run_plugin ($st, NBP, STATUSMETHOD);
-    }
+    my ($self, $node, $st) = @_;
+
+    $self->debug (1, "Showing the state of $node");
+    $self->run_plugin ($st, NBP, STATUSMETHOD);
 }
 
 # Runs the Boot method of the NBP plugins of the nodes given as
 # arguments.
-sub boot
+sub _boot
 {
-    my ($self, %nl) = @_;
-
-    while (my ($node, $st) = each (%nl)) {
-        $self->debug (1, "Booting: $node");
-        my $lock = $self->lock_node($node);
-        if (! $lock) {
-            $self->error("aii-shellfe: couldn't acquire lock on $node for boot");
-            exit(1);
-        }
-        $self->run_plugin ($st, NBP, BOOTMETHOD);
-    }
+    my ($self, $node, $st) = @_;
+    $self->run_plugin ($st, NBP, BOOTMETHOD);
 }
 
 # Runs the Firmware method of the NBP plugins of the nodes given as
 # arguments.
-sub firmware
+sub _firmware
 {
-    my ($self, %nl) = @_;
-
-    while (my ($node, $st) = each (%nl)) {
-        $self->debug (1, "Updating firmware of $node");
-        my $lock = $self->lock_node($node);
-        if (! $lock) {
-            $self->error("aii-shellfe: couldn't acquire lock on $node for firmware");
-            exit(1);
-        }
-        $self->run_plugin ($st, NBP, FIRMWAREMETHOD);
-    }
+    my ($self, $node, $st) = @_;
+    $self->run_plugin ($st, NBP, FIRMWAREMETHOD);
 }
 
 # Runs the Livecd method of the NBP plugins of the nodes given as
 # arguments
-sub livecd
+sub _livecd
 {
-    my ($self, %nl) = @_;
-    while (my ($node, $st) = each (%nl))  {
-        $self->debug(1, "Setting to livecd: $node");
-        my $lock = $self->lock_node($node);
-        if (! $lock) {
-            $self->error("aii-shellfe: couldn't acquire lock on $node for livecd");
-            exit(1);
-        }
-        $self->run_plugin($st, NBP, LIVECDMETHOD);
-    }
+    my ($self, $node, $st) = @_;
+    $self->run_plugin($st, NBP, LIVECDMETHOD);
 }
 
 # Runs the Remove method of the NBP plugins of the nodes given as
 # arguments.
-sub remove
+sub _remove
 {
-    my ($self, %nl) = @_;
-
-    while (my ($node, $st) = each (%nl)) {
-        $self->debug (1, "Removing $node");
-        my $lock = $self->lock_node($node);
-        if (! $lock) {
-            $self->error("aii-shellfe: couldn't acquire lock on $node for remove");
-            exit(1);
-        }
+    my ($self, $node, $st) = @_;
+
+    $self->iter_plugins($st, REMOVEMETHOD);
```

Comment:
certainly not intentional. it's a result from the interactive rebase. splitting commits is not that trivial. i'll try to fix it